Sources/Variable.swift

```diff
@@ -78,36 +78,44 @@ public struct Variable : Equatable, Resolvable {
     for bit in lookup() {
       current = normalize(current)
 
+      // try to evaluate bit if it's an indirection
+      var resolvedBit = bit
+      if bit.hasPrefix("$"),
+        let resolved = try? Variable(String(bit.dropFirst())).resolve(context),
+        let property = resolved {
```

Comment:
maybe better to do `resolved as? String`?

Reply:
I had that first, but things break then if the resolved value is an `Int` for example (see the tests with array indirect lookup)